<?xml version="1.0" encoding="UTF-8"?>
<testng-results ignored="0" total="7" passed="7" failed="0" skipped="0">
  <reporter-output>
  </reporter-output>
  <suite started-at="2021-10-16T15:44:57 IST" name="Default suite" finished-at="2021-10-16T15:44:57 IST" duration-ms="104">
    <groups>
    </groups>
    <test started-at="2021-10-16T15:44:57 IST" name="Default test" finished-at="2021-10-16T15:44:57 IST" duration-ms="104">
      <class name="calculator.StringCalculatorShould">
        <test-method signature="empty_string_should_return_0()[pri:0, instance:calculator.StringCalculatorShould@7b227d8d]" started-at="2021-10-16T15:44:57 IST" name="empty_string_should_return_0" finished-at="2021-10-16T15:44:57 IST" duration-ms="16" status="PASS">
          <reporter-output>
          </reporter-output>
        </test-method> <!-- empty_string_should_return_0 -->
        <test-method signature="ignoresNumbersGreaterThan1000()[pri:0, instance:calculator.StringCalculatorShould@7b227d8d]" started-at="2021-10-16T15:44:57 IST" name="ignoresNumbersGreaterThan1000" finished-at="2021-10-16T15:44:57 IST" duration-ms="3" status="PASS">
          <reporter-output>
          </reporter-output>
        </test-method> <!-- ignoresNumbersGreaterThan1000 -->
        <test-method signature="negativeInputsReturnException()[pri:0, instance:calculator.StringCalculatorShould@7b227d8d]" started-at="2021-10-16T15:44:57 IST" name="negativeInputsReturnException" finished-at="2021-10-16T15:44:57 IST" duration-ms="3" status="PASS">
          <exception class="java.lang.Exception">
            <message>
              <![CDATA[negatives not allowed]]>
            </message>
            <full-stacktrace>
              <![CDATA[java.lang.Exception: negatives not allowed
at calculator.StringCalculator.findDengerousInput(StringCalculator.java:35)
at calculator.StringCalculator.getSum(StringCalculator.java:18)
at calculator.StringCalculator.add(StringCalculator.java:14)
at calculator.StringCalculatorShould.negativeInputsReturnException(StringCalculatorShould.java:37)
at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
at java.base/java.lang.reflect.Method.invoke(Method.java:564)
at org.testng.internal.MethodInvocationHelper.invokeMethod(MethodInvocationHelper.java:133)
at org.testng.internal.TestInvoker.invokeMethod(TestInvoker.java:598)
at org.testng.internal.TestInvoker.invokeTestMethod(TestInvoker.java:173)
at org.testng.internal.MethodRunner.runInSequence(MethodRunner.java:46)
at org.testng.internal.TestInvoker$MethodInvocationAgent.invoke(TestInvoker.java:824)
at org.testng.internal.TestInvoker.invokeTestMethods(TestInvoker.java:146)
at org.testng.internal.TestMethodWorker.invokeTestMethods(TestMethodWorker.java:146)
at org.testng.internal.TestMethodWorker.run(TestMethodWorker.java:128)
at java.base/java.util.ArrayList.forEach(ArrayList.java:1511)
at org.testng.TestRunner.privateRun(TestRunner.java:794)
at org.testng.TestRunner.run(TestRunner.java:596)
at org.testng.SuiteRunner.runTest(SuiteRunner.java:377)
at org.testng.SuiteRunner.runSequentially(SuiteRunner.java:371)
at org.testng.SuiteRunner.privateRun(SuiteRunner.java:332)
at org.testng.SuiteRunner.run(SuiteRunner.java:276)
at org.testng.SuiteRunnerWorker.runSuite(SuiteRunnerWorker.java:53)
at org.testng.SuiteRunnerWorker.run(SuiteRunnerWorker.java:96)
at org.testng.TestNG.runSuitesSequentially(TestNG.java:1212)
at org.testng.TestNG.runSuitesLocally(TestNG.java:1134)
at org.testng.TestNG.runSuites(TestNG.java:1063)
at org.testng.TestNG.run(TestNG.java:1031)
at org.testng.remote.AbstractRemoteTestNG.run(AbstractRemoteTestNG.java:115)
at org.testng.remote.RemoteTestNG.initAndRun(RemoteTestNG.java:251)
at org.testng.remote.RemoteTestNG.main(RemoteTestNG.java:77)
]]>
            </full-stacktrace>
          </exception> <!-- java.lang.Exception -->
          <reporter-output>
          </reporter-output>
        </test-method> <!-- negativeInputsReturnException -->
        <test-method signature="string_with_single_number_should_return_number_as_int()[pri:0, instance:calculator.StringCalculatorShould@7b227d8d]" started-at="2021-10-16T15:44:57 IST" name="string_with_single_number_should_return_number_as_int" finished-at="2021-10-16T15:44:57 IST" duration-ms="1" status="PASS">
          <reporter-output>
          </reporter-output>
        </test-method> <!-- string_with_single_number_should_return_number_as_int -->
        <test-method signature="threeNumbersDeliminatedBothWaysReturnSum()[pri:0, instance:calculator.StringCalculatorShould@7b227d8d]" started-at="2021-10-16T15:44:57 IST" name="threeNumbersDeliminatedBothWaysReturnSum" finished-at="2021-10-16T15:44:57 IST" duration-ms="2" status="PASS">
          <reporter-output>
          </reporter-output>
        </test-method> <!-- threeNumbersDeliminatedBothWaysReturnSum -->
        <test-method signature="twoNumbersCommaDeliminatedReturnSum()[pri:0, instance:calculator.StringCalculatorShould@7b227d8d]" started-at="2021-10-16T15:44:57 IST" name="twoNumbersCommaDeliminatedReturnSum" finished-at="2021-10-16T15:44:57 IST" duration-ms="3" status="PASS">
          <reporter-output>
          </reporter-output>
        </test-method> <!-- twoNumbersCommaDeliminatedReturnSum -->
        <test-method signature="twoNumbersSeprateLineDeliminatedReturnSum()[pri:0, instance:calculator.StringCalculatorShould@7b227d8d]" started-at="2021-10-16T15:44:57 IST" name="twoNumbersSeprateLineDeliminatedReturnSum" finished-at="2021-10-16T15:44:57 IST" duration-ms="1" status="PASS">
          <reporter-output>
          </reporter-output>
        </test-method> <!-- twoNumbersSeprateLineDeliminatedReturnSum -->
      </class> <!-- calculator.StringCalculatorShould -->
    </test> <!-- Default test -->
  </suite> <!-- Default suite -->
</testng-results>
